statsmodels/miscmodels/betareg.py

```diff
@@ -0,0 +1,465 @@
+                    if hasattr(exog_precision, 'columns')
+                    else range(1, exog_precision.shape[1] + 1))]
+
+        kwds['extra_params_names'] = extra_names
+
+        super(Beta, self).__init__(endog, exog, **kwds)
+        self.link = link
+        self.link_precision = link_precision
+
+        self.exog_precision = exog_precision
+        assert len(self.exog_precision) == len(self.endog)
+
+    @classmethod
+    def from_formula(cls, formula, data, exog_precision_formula=None,
+                     *args, **kwargs):
+        if exog_precision_formula is not None:
+            if 'subset' in kwargs:
+                d = data.ix[kwargs['subset']]
+                Z = patsy.dmatrix(exog_precision_formula, d)
+            else:
+                Z = patsy.dmatrix(exog_precision_formula, data)
+            kwargs['exog_precision'] = Z
+
+        return super(Beta, cls).from_formula(formula, data, *args,
+                                      **kwargs)
+
+
+    def predict(self, params, exog=None):
+        """predict values for mean, conditional expectation E(endog | exog)
+
+        """
+        if exog is None:
+            exog = self.exog
+        k_mean = self.exog.shape[1]
+
+        params_mean = params[:k_mean]
+        Zparams = params[k_mean:]
+        mu = self.link.inverse(np.dot(exog, params_mean))
+        return mu
+
+
+    def predict_precision(self, params, exog_precision=None):
+        """predict values for precision parameter for given exog_precision
+
+        """
+        if exog_precision is None:
+            exog_precision = self.exog_precision
+
+        k_mean = self.exog.shape[1]
+        params_precision = params[k_mean:]
+        linpred = np.dot(exog_precision, params_precision)
+        phi = self.link_precision.inverse(linpred)
+
+        return phi
+
+
+    def predict_var(self, params, exog=None, exog_precision=None):
+        """predict values for conditional variance V(endog | exog)
+
+        """
+        mean = self.predict(params, exog=exog)
+        precision = self.predict_precision(params,
+                                           exog_precision=exog_precision)
+
+        var_endog = mean * (1 - mean) / (1 + precision)
+        return var_endog
+
+
+    def nloglikeobs(self, params):
+        """
+        Negative log-likelihood.
+
+        Parameters
+        ----------
+
+        params : np.ndarray
+            Parameter estimates
+        """
+        return -self._ll_br(self.endog, self.exog, self.exog_precision, params)
+
+    def _ll_br(self, y, X, Z, params):
+        nz = Z.shape[1]
+
+        Xparams = params[:-nz]
+        Zparams = params[-nz:]
+
+        mu = self.link.inverse(np.dot(X, Xparams))
+        phi = self.link_precision.inverse(np.dot(Z, Zparams))
+
+        alpha = mu * phi
+        beta = (1 - mu) * phi
+
+        if np.any(alpha <= np.finfo(float).eps): return np.array(-np.inf)
+        if np.any(beta <= np.finfo(float).eps): return np.array(-np.inf)
```

Comment:
is this in place of a `clip`?

both mu and phi are positive with default links
I guess this only handles the case >=0 and <= eps, essentially =0.